MODULE.bazel

```diff
@@ -0,0 +1,197 @@
+module(
+    name = "grpc-java",
+    version = "0",
+    compatibility_level = 0,
+    repo_name = "io_grpc_grpc_java",
+)
+
+bazel_dep(name = "googleapis", version = "0.0.0-20240326-1c8d509c5", repo_name = "com_google_googleapis")
+bazel_dep(name = "grpc", version = "1.56.3.bcr.1", repo_name = "com_github_grpc_grpc")
+bazel_dep(name = "protobuf", version = "23.1", repo_name = "com_google_protobuf")
+bazel_dep(name = "rules_cc", version = "0.0.9")
+bazel_dep(name = "rules_go", version = "0.46.0", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "rules_jvm_external", version = "6.0")
+bazel_dep(name = "rules_proto", version = "5.3.0-21.7")
+
+non_module_deps = use_extension("//:repositories.bzl", "grpc_java_repositories_extension")
+use_repo(
+    non_module_deps,
+    "com_github_cncf_xds",
+    "envoy_api",
+    "io_grpc_grpc_proto",
+)
+
+grpc_repo_deps_ext = use_extension("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_repo_deps_ext")
+use_repo(
+    grpc_repo_deps_ext,
+    "com_envoyproxy_protoc_gen_validate",
+    "com_github_cncf_udpa",
+    "opencensus_proto",
+)
+
+maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
+maven.install(
+    name = "grpc_java_maven",
```

Comment:
Looking at [the implementation](https://github.com/bazelbuild/rules_jvm_external/blob/0ef17e79e033832f9239b8c2a1122dbe6f8ff720/private/extensions/maven.bzl#L201) it seems we should use the default name (`maven`) to allow these artifacts to be combined with all other modules and a single version chosen for each. We also _shouldn't_ have the lock file because only one module can define that, and we want that to be the grpc-user's module.

Reply:
Done, originally I was thinking the separate repo made sense, but you're right that users should take the versions from this repo into account